src/main/java/com/example/git_repo/error/decoder/GitErrorDecoder.java

```diff
@@ -0,0 +1,18 @@
+package com.example.git_repo.error.decoder;
+
+import feign.Response;
+import feign.codec.ErrorDecoder;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.server.ResponseStatusException;
+
+public class GitErrorDecoder implements ErrorDecoder {
+
+    @Override
+    public Exception decode(String methodKey, Response response) {
+        if (response.status() == HttpStatus.SERVICE_UNAVAILABLE.value()) {
```

Comment:
Niebezpieczne zastosowanie. Moim zdaniem na koncu powinienes nie zwraca new Exception Generic Error bo to nam nic nie mowi tylko moze lepiej na ten moment przekierowac do Defaultowego Error Decodera. Dzieki czemu dodajemy to co nas interesuje dodac a wszystkie inne przypadki beda wpadaly w sciezce defaultowa